lib/chunks.sql

```diff
@@ -1,119 +1,222 @@
-    SELECT tq.id, tq.qid, tq.course_id, tq.template_directory
+    SELECT
+      tq.id,
+      tq.qid,
+      tq.course_id,
+      tq.template_directory
     FROM
-        template_questions AS q
-        JOIN questions AS tq ON (tq.qid = q.template_directory AND tq.course_id = q.course_id)
-    WHERE tq.deleted_at IS NULL
-)
-SELECT id FROM template_questions LIMIT 100; -- LIMIT prevents infinite recursion on circular templates
+      template_questions AS q
+      JOIN questions AS tq ON (
+        tq.qid = q.template_directory
+        AND tq.course_id = q.course_id
+      )
+    WHERE
+      tq.deleted_at IS NULL
+  )
+SELECT
+  id
+FROM
+  template_questions
+LIMIT
+  100;
 
+-- LIMIT prevents infinite recursion on circular templates
 -- BLOCK insert_chunks
-INSERT INTO chunks (type, course_id, question_id, course_instance_id, assessment_id, uuid)
+INSERT INTO
+  chunks (
+    type,
+    course_id,
+    question_id,
+    course_instance_id,
+    assessment_id,
+    uuid
+  )
 SELECT
-    (chunk->>'type')::enum_chunk_type,
-    $course_id,
-    q.id,
-    ci.id,
-    a.id,
-    (chunk->>'uuid')::uuid
+  (chunk ->> 'type')::enum_chunk_type,
+  $course_id,
+  q.id,
+  ci.id,
+  a.id,
+  (chunk ->> 'uuid')::uuid
 FROM
-    JSON_ARRAY_ELEMENTS($chunks) AS chunk(json)
-    LEFT JOIN questions AS q ON (q.qid = (chunk->>'questionName') AND q.deleted_at IS NULL AND q.course_id = $course_id)
-    LEFT JOIN course_instances AS ci ON (ci.short_name = (chunk->>'courseInstanceName') AND ci.deleted_at IS NULL AND ci.course_id = $course_id)
-    LEFT JOIN assessments AS a ON (a.tid = (chunk->>'assessmentName') AND a.deleted_at IS NULL AND a.course_instance_id = ci.id)
-ON CONFLICT (type, course_id, coalesce(question_id, -1), coalesce(course_instance_id, -1), coalesce(assessment_id, -1)) DO UPDATE
+  JSON_ARRAY_ELEMENTS($chunks) AS chunk (json)
+  LEFT JOIN questions AS q ON (
+    q.qid = (chunk ->> 'questionName')
+    AND q.deleted_at IS NULL
+    AND q.course_id = $course_id
+  )
+  LEFT JOIN course_instances AS ci ON (
+    ci.short_name = (chunk ->> 'courseInstanceName')
+    AND ci.deleted_at IS NULL
+    AND ci.course_id = $course_id
+  )
+  LEFT JOIN assessments AS a ON (
+    a.tid = (chunk ->> 'assessmentName')
+    AND a.deleted_at IS NULL
+    AND a.course_instance_id = ci.id
+  )
+ON CONFLICT (
+  type,
+  course_id,
+  coalesce(question_id, -1),
+  coalesce(course_instance_id, -1),
+  coalesce(assessment_id, -1)
+) DO
+UPDATE
 SET
-    type = EXCLUDED.type,
-    question_id = EXCLUDED.question_id,
-    course_instance_id = EXCLUDED.course_instance_id,
-    assessment_id = EXCLUDED.assessment_id,
-    uuid = EXCLUDED.uuid;
+type = EXCLUDED.type,
+question_id = EXCLUDED.question_id,
+course_instance_id = EXCLUDED.course_instance_id,
+assessment_id = EXCLUDED.assessment_id,
+uuid = EXCLUDED.uuid;
 
 -- BLOCK delete_chunks
-WITH chunks_metadata AS (
+WITH
+  chunks_metadata AS (
+    SELECT
+      (cm ->> 'type')::enum_chunk_type AS
```

Comment:
It'll take me some time to adjust to spaces on either side of JSON operators 😅

Reply:
`sql-formatter` has an option to remove those spaces, but it's all-or-nothing, and setting it would also change operators like `+`.

Reply:
Understood - I think spaces around operators is on the whole the way to go.